source/extended-json.rst

```diff
@@ -0,0 +1,618 @@
+      },
+     "Binary": {
+         "$binary": "o0w498Or7cijeBSpkquNtg==",
+         "$type": "03"
+     },
+     "BinaryUserDefined": {
+         "$binary": "AQIDBAU=",
+         "$type": "80"
+     },
+     "Code": {
+         "$code": "function() {}"
+     },
+     "CodeWithScope": {
+         "$code": "function() {}",
+         "$scope": {}
+     },
+     "Subdocument": {
+         "foo": "bar"
+     },
+     "Array": [
+          {"$numberInt": "1"},
+          {"$numberInt": "2"},
+          {"$numberInt": "3"},
+          {"$numberInt": "4"},
+          {"$numberInt": "5"}
+      ],
+     "Timestamp": {
+         "$timestamp": "180388626433"
+      },
+     "Regex": {
+         "$regex": "pattern",
+         "$options": ""
+     },
+     "DatetimeEpoch": {
+         "$date": {
+             "$numberLong": "0"
+         }
+     },
+     "DatetimePositive": {
+         "$date": {
+             "$numberLong": "9223372036854775807"
+         }
+     },
+     "DatetimeNegative": {
+         "$date": {
+             "$numberLong": "-9223372036854775808"
+         }
+     },
+     "True": true,
+     "False": false,
+     "DBPointer": {
+          "$dbPointer": {
+             "$ref": "db.collection",
+             "$id": {
+                 "$oid": "57e193d7a9cc81b4027498b1"
+             }
+          }
+     },
+      "DBRef": {
+          "$ref": "collection",
+          "$id": {
+              "$oid": "57fd71e96e32ab4225b723fb"
+          },
+          "$db": "database"
+      },
+     "Minkey": {
+         "$minKey": 1
+     },
+     "Maxkey": {
+         "$maxKey": 1
+     },
+     "Null": null,
+     "Undefined": {
+          "$undefined": true
+      }
+  }
+
+Test Plan
+=========
+
+Drivers, tools, and libraries can test their compliance to this specification by
+running the tests in version 1.3 and above of the `BSON Corpus Test Suite`_.
+
+.. _BSON Corpus Test Suite: https://github.com/mongodb/specifications/blob/master/source/bson-corpus/bson-corpus.rst
+
+Motivation for Change
+=====================
+
+There existed many extended JSON parser and generator implementations prior to
+this specification that used conflicting formats, since there was no agreement
+on the precise format of extended JSON. This resulted in problems where the
+output of some generators could not be consumed by some parsers.
+
+Design Rationale
+================
+
+Of Parsers and Generators
+-------------------------
+
+Parsers MUST accept all strings in canonical extended JSON, and generators MUST
+have the ability to output canonical extended JSON. This way, a parser will
+always be able to accept strings emitted by a properly configured
+generator. Parsers and generators are permitted to accept and output strings in
+other formats as well for backwards compatibility.
+
+Generators are  required to  support at least  100 levels of  nesting in  a BSON
+document  being transformed  to extended  JSON. This  aligns with  MongoDB’s own
```

Comment:
Quite a bit of extra whitespace on this line.

Also, you're using the `’` character instead of `'` here.